ibm/resource_ibm_kms_key_rings.go

```diff
@@ -0,0 +1,248 @@
+package ibm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+
+	kp "github.com/IBM/keyprotect-go-client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceIBMKmskeyRings() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMKmsKeyRingCreate,
+		Delete:   resourceIBMKmsKeyRingDelete,
+		Read:     resourceIBMKmsKeyRingRead,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Key ID",
+				ForceNew:    true,
+			},
+			"key_ring_id": {
```

Comment:
API DOcs has validation fo rthis attribute..
Please add the validation  `Constraints: 2 ≤ length ≤ 100, Value must match regular expression ^[a-zA-Z0-9-]*$`
Use new way of validation.. Refer [validation](https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/resource_ibm_cr_namespace.go#L30)

Reply:
Need Discussion